ls1intum/Artemis
# `Assessment`: Show number of filtered entries in scores table

### Checklist
#### General


#### Client

### Motivation and Context


When instructors take a look at the scores table and especially the filters there, they want to know how many scores are affected by each filter and how many scores there are in total. The scores table should show this information in an clear way to the user.

### Description

Displays next to each filter the number of scores that fit into this filter. Additionally shows how many scores there are in total when some are filtered out

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with some scores

1. Log in to Artemis
2. Participate with some students or find an exercise with different participations. Some successful, some with errors, some with a build failure or a locked assessment, ...
3. Go to the scores table of that exercise and look at the filter.
4. Make sure that the number displayed next to the filter is correct.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [exercise-scores.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5061-1/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/exercise-scores/exercise-scores.component.ts.html) | 88.23% | ✅ |

 

### Screenshots


![Bildschirmfoto 2023-06-29 um 10 38 33](https://github.com/ls1intum/Artemis/assets/38322605/9fddf435-41e6-4915-96d0-802c88f8bbc2)
